### PR TITLE
Update configuration tests with per library keys, JSON type values

### PR DESCRIPTION
**Pull request by Kyle-Verhoog**
We're permitting that types be sent as JSON in addition to strings so that the backend can normalize values like `null` consistently. Note that the telemetry platform [**does not support** objects and dictionary values in payloads](https://github.com/DataDog/instrumentation-telemetry-api-docs/blob/main/GeneratedDocumentation/ApiDocs/v2/SchemaDocumentation/Schemas/conf_key_value.md).

Other changes include supporting "inactive" for appsec_enabled which is supported in Java and indicates that the library is currently disabled but can be enabled using Remote Config.

Also, we're going to relax the tests to accept each key that the libraries wish to send and hope that the resulting mapping is implemented in the backend.

## Reviewer checklist

* [ ] [Relevant label](https://github.com/DataDog/system-tests/blob/main/docs/CI/labels.md) (`run-parametric-scenario`, `run-profiling-scenario`...) are presents
* [ ] No system-tests internal is modified. Otherwise, I have the approval from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)
* [ ] CI is green, or failing jobs are not related to this change (and you are 100% sure about this statement)
* [ ] A docker base image is modified?
    * [ ] the relevant `build-XXX-image` label is present
    * [ ] To R&P team: locally build and push the image to hub.docker.com 
* [ ] A scenario is added (or removed)?
    * [ ] Get a review from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)
    * [ ] Once merged, add (or remove) it in system-test-dasboard nightly

### CODE REVIEW

File: tests/parametric/test_telemetry.py [hunk 1]
```diff
@@ -83,21 +86,32 @@ def test_library_settings(self, library_env, test_agent, test_library):
...
-            ("trace_tags", "team:apm,component:web"),
-            ("trace_enabled", "true"),
-            ("profiling_enabled", "false"),
-            ("appsec_enabled", "false"),
-            ("data_streams_enabled", "false"),
+            ("trace_sample_rate", ("0.3", 0.3)),
+            ("logs_injection_enabled", ("true", True)),
+            (
+                "trace_header_tags",
+                ("X-Header-Tag-1:header_tag_1,X-Header-Tag-2:header_tag_2"),
```
Comment (ygree) [inline tests/parametric/test_telemetry.py:93]:
```suggestion
                ("X-Header-Tag-1:header_tag_1,X-Header-Tag-2:header_tag_2",
                {"X-Header-Tag-1": "header_tag_1", "X-Header-Tag-2": "header_tag_2"}),
```

File: tests/parametric/test_telemetry.py [hunk 2]
```diff
@@ -83,21 +86,32 @@ def test_library_settings(self, library_env, test_agent, test_library):
...
+                {"X-Header-Tag-1": "header_tag_1", "X-Header-Tag-2": "header_tag_2"},
+            ),
+            ("trace_tags", ("team:apm,component:web", {"team": "apm", "component": "web"})),
```
Comment (ygree) [inline tests/parametric/test_telemetry.py:96]:
```suggestion
            ("trace_tags", ("team:apm,component:web", "component:web,team:apm", {"team": "apm", "component": "web"})),
```

Add a case for Java

File: tests/parametric/test_telemetry.py [hunk 3]
```diff
@@ -34,21 +34,24 @@ def test_library_settings(self, library_env, test_agent, test_library):
...
         for (apm_telemetry_name, value) in [
-            ("trace_sample_rate", "1.0"),
-            ("logs_injection_enabled", "false"),
-            ("trace_header_tags", ""),
-            ("trace_tags", ""),
-            ("trace_enabled", "true"),
-            ("profiling_enabled", "false"),
-            ("appsec_enabled", "false"),
-            ("data_streams_enabled", "false"),
+            ("trace_sample_rate", (1.0, None, "1.0")),
```
Comment (ygree) [inline tests/parametric/test_telemetry.py:37]: Nice, I like simplicity of this approach to allow different matching values!

Comment (ygree) [reply tests/parametric/test_telemetry.py:37]: BTW, does `None` in Python corresponds to `null` in JSON?

File: tests/parametric/test_telemetry.py [hunk 4]
```diff
@@ -11,6 +11,26 @@
 from utils import context, scenarios, rfc, features
 
 
+telemetry_name_mapping = {
+    "trace_sample_rate": {"dotnet": "TODO",},
```
Comment (Kyle-Verhoog) [inline tests/parametric/test_telemetry.py:15]: @kevingosse can you fill these in for .NET? 🙏

Comment (kevingosse) [reply tests/parametric/test_telemetry.py]: Done in https://github.com/DataDog/system-tests/pull/2172/commits/0f6af6416d1169f9541afad1888aeb6afc7343cd

**Review by avivenzio-dd — APPROVED** [review]

**Review by robertomonteromiguel — APPROVED** [review]

File: manifests/nodejs.yml [hunk 5]
```diff
@@ -403,8 +403,8 @@ tests/:
       TestDynamicConfigV2: v4.23.0
     test_span_links.py: missing_feature
     test_telemetry.py:
-      Test_Defaults: missing_feature
-      Test_Environment: missing_feature
+      Test_Defaults: 
+      Test_Environment:
```
Comment (simon-id) [inline manifests/nodejs.yml:407]: uh ?

File: tests/parametric/test_telemetry.py [hunk 6]
```diff
@@ -69,6 +94,7 @@ class Test_Environment:
                 "DD_TRACE_HEADER_TAGS": "X-Header-Tag-1:header_tag_1,X-Header-Tag-2:header_tag_2",
                 "DD_TAGS": "team:apm,component:web",
                 "DD_TRACE_ENABLED": "true",
+                "DD_TRACING_ENABLED": "true",
```
Comment (Kyle-Verhoog) [inline tests/parametric/test_telemetry.py:97]:
```suggestion
                # node.js DD_TRACING_ENABLED is equivalent to DD_TRACE_ENABLED in other libraries
                "DD_TRACING_ENABLED": "true",
```

**Pull request merged**